### PR TITLE
Add articles full BE w/tests, FE model, card, gallery

### PR DESCRIPTION
**Pull request by kyunoske**

### CODE REVIEW

File: backend/src/test/java/com/example/backend/controller/ArticleControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    MockMvc mockMvc;
+
+    @Autowired
+    private ArticleRepo articleRepo;
+
+    @MockBean
+    private IdService idService;
+
+    @Test
+    void getAllArticles() throws Exception {
```
Comment (Kerambad) [inline backend/src/test/java/com/example/backend/controller/ArticleControllerTest.java:37]: Maybe consider using a more specific name for testing method ;)

File: frontend/src/components/AddTrackModal.tsx [hunk 2]
```diff
@@ -45,10 +45,6 @@ function AddTrackModal(props: AddTrackModalProps) {
             props.addTrack(track);
             console.table(track)
         }
-
-        setTimeout(() => {
-            console.log('Hello, World!')
```
Comment (Kerambad) [inline frontend/src/components/AddTrackModal.tsx:50]: Is this log supposed to get logged in the final app?

File: frontend/src/hooks/useArticles.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    const getAllArticles = () => {
+        axios.get("/api/articles")
+            .then((response) => response.data)
+            .then((articles) => setArticles(articles))
+            .catch((error) => console.log(error))
+    }
+
+    const getArticleById = (id: string) => {
+        axios.get(`/api/articles/${id}`)
+            .then(response => response.data);
```
Comment (Kerambad) [inline frontend/src/hooks/useArticles.tsx:24]: Consider catch errors with all axios requests

**Review by Kerambad — APPROVED** [review]
Looks good!

File: frontend/src/components/AddTrackModal.tsx (hunk 2 above)
Comment (kyunoske) [reply frontend/src/components/AddTrackModal.tsx:50]: I forgot to delete that.

File: frontend/src/hooks/useArticles.tsx (hunk 3 above)
Comment (kyunoske) [reply frontend/src/hooks/useArticles.tsx:24]: Ok I will add that, thanks Marek!

**Pull request merged**